File: javascript/125. Valid Palindrome.js
var isPalindrome = function(s) {
    const isNumber = s => s.charCodeAt() >= 48 && s.charCodeAt() <= 57
    const isLetter = s => s.charCodeAt() >= 97 && s.charCodeAt() <= 122
    let lowercase = s.toLowerCase()
    
    function _removeNonAlphanumeric(s) {
        return s.slice().split('').filter(word => isNumber(word) || isLetter(word)).join('')
    }
    
    let onlyAlphaOrNum = _removeNonAlphanumeric(lowercase)
    if(onlyAlphaOrNum.length === 0 || onlyAlphaOrNum.length === 1) return true
    let left = 0, right = onlyAlphaOrNum.length - 1
    while(left <= right) {
        if(onlyAlphaOrNum[left] !== onlyAlphaOrNum[right]) return false;
        left ++;
        right --;
    }
    
    return true;
};

File: java/125. Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        if (s.length() == 1) return true;

        String[] splitted = s.split("");
        String onlyAlpha = "";

        for(String c : splitted) {
            String lowerC = c.toLowerCase();
            int codePointC = lowerC.codePointAt(0);
            if((codePointC >= 48 && codePointC <= 57) || (codePointC >= 97 && codePointC <= 122)) {
                onlyAlpha += lowerC;
            }
        }

        if (onlyAlpha.length() <= 1) return true;

        int left = 0, right = onlyAlpha.length() - 1;

        while(left <= right) {
            if(onlyAlpha.charAt(left) != onlyAlpha.charAt(right)) return false;
            left++;
            right--;
        }

        return true;
    }
}